File: calendars/calendar_day_update.go
package calendars

type CalendarDayUpdate struct {
	Condition          CurrentMarketCondition `json:"condition"`
	At                 int64                  `json:"at"`
	PreviousMarketDate CalendarDay            `json:"prev"`
	CurrentMarketDate  CalendarDay            `json:"current"`
	NextMarketDate     CalendarDay            `json:"next"`
}

File: frontend/src/types/CalendarDayUpdate.ts
import { CalendarDay } from './CalendarDay'
import { CurrentMarketCondition } from '@/types/CurrentMarketCondition'

export interface CalendarDayUpdate {
  condition: CurrentMarketCondition
  at: number
  prev: CalendarDay
  current: CalendarDay
  next: CalendarDay
}
